master/internal/api_experiment.go

```diff
@@ -1719,6 +1719,33 @@ func (a *apiServer) CreateExperiment(
 	}, nil
 }
 
+func (a *apiServer) PutExperimentRetainLogs(
+	ctx context.Context, req *apiv1.PutExperimentRetainLogsRequest,
+) (*apiv1.PutExperimentRetainLogsResponse, error) {
+	results, err := experiment.RetainLogs(ctx, a.m.db, []int32{req.ExperimentId}, nil, int16(req.NumDays))
+	if err != nil {
+		return nil, err
+	}
+	if err == nil {
+		if len(results) == 0 {
+			return nil, errors.Errorf("MoveExperiments returned neither pass nor fail on query")
+		} else if results[0].Error != nil {
+			return nil, results[0].Error
+		}
+	}
```

Comment:
```suggestion
	if len(results) == 0 {
		return nil, errors.Errorf("MoveExperiments returned neither pass nor fail on query")
	} else if results[0].Error != nil {
		return nil, results[0].Error
	}
```
we know the error is nil here already so we don't need this if. also, i prefer a switch in cases like this.

Reply:
Done!